### ISSUE TITLE
Cleanup Views support in Locale module - remove broken handlers for textgroup

### ISSUE THREAD

**Issue by indigoxela** [backdrop/backdrop-issues#4838]
**Description of the bug**

1. Field "textgroup" has been removed from core, but locale.views.inc still uses it
2. Native language display is gone - (property removed from language object) but views still uses it

After some forward and back the consensus is to let contrib module i18n handle textgroups and remove all broken remnants from core. See [this comment](https://github.com/backdrop/backdrop-issues/issues/4838#issuecomment-778611668) for details.

The problem with the native language property has its own issue (the property gets restored).

<hr />

**Steps To Reproduce Problem 1**

1. Enable locale, language modules
2. Create a view of type "Locale source"
3. Add a field "Locale source: Group", save and refresh preview

You get:

`SQLSTATE[42S22]: Column not found: 1054 Unknown column 'locales_source.textgroup' in 'field list'`

Reason: Textgroups have been removed from core.

**Steps To Reproduce Problem 2**

(Addressed in https://github.com/backdrop/backdrop-issues/issues/4511)

1. After step 1 and 2
2. Add a field "Locale target: Language", set display to "Native language", save and refresh preview

You get a bunch of:

`Notice: Undefined property: stdClass::$native in views_language_list() (line 752 of .../core/modules/views/views.module).`

Reason: The "native" property has been removed from the $language object

**Additional information**

Re textgroups see this [Drupal issue](https://www.drupal.org/project/drupal/issues/1188430)
And also: https://github.com/backdrop-contrib/i18n/issues/17 (i18n contrib module struggles with this, too).

Re native property see this [related Issue comment](https://github.com/backdrop/backdrop-issues/issues/4511#issuecomment-667667236)

We could add back both, textgroups and native property to core or we should remove their usage in views. Whatever we do to fix both things - views should not throw Exceptions or Notices.

**Comment by indigoxela** [comment]
In theory #4511 fixes the "native language name" part of this issue. I didn't test, though.

**Comment by herbdool** [comment]
I'm leaning towards bringing them both back.

**Comment by indigoxela** [comment]
> I'm leaning towards bringing them both back.

@herbdool are you willing to test and/or review that related [issue/PR](https://github.com/backdrop/backdrop-issues/issues/4511)?

**Comment by herbdool** [comment]
@indigoxela I've created a PR assuming that we only support textgroup in i18n.

**Comment by indigoxela** [comment]
Unfortunately this issue here is (sort of) blocked by [another core issue](https://github.com/backdrop/backdrop-issues/issues/4511)

**Comment by herbdool** [comment]
@indigoxela I copied the bug report to that other issue since they can be treated separately. So this issue is just about `textgroup`. And the PR looks good.

**Comment by indigoxela** [comment]
Ehm.... ouch! I don't think that the one who provides the PR is supposed to set it to RTBC.

An I'm not sure about your approach yet. Does something like `module_exists('i18n')` really belong in core? Wouldn't it be cleaner, if i18n did a [hook_views_data_alter()](https://docs.backdropcms.org/api/backdrop/core%21modules%21views%21views.api.php/function/hook_views_data_alter/1)?

**Comment by herbdool** [comment]
@indigoxela oops! In my hurry I forgot I added the PR.

If you want to take that approach that's fine. I'll wait to see if your PR on i18n works before closing this one.

**Comment by indigoxela** [comment]
A PR for the contrib module i18n has been created.

Now let's fix this in core. We need to remove:

- `$data['locales_source']['textgroup']` from locale_views_data()
- File core/modules/locale/views/views_handler_argument_locale_group.inc
- File core/modules/locale/views/views_handler_field_locale_group.inc
- File core/modules/locale/views/views_handler_filter_locale_group.inc
- The definitions of above three handlers in core/modules/locale/views/locale.views.inc
- The autoload info of above three handlers in core/modules/locale/locale.module (locale_autoload_info)

It turned out that all handlers were broken not only because of the missing table column, but also because they all still used `hook_locale`, which got removed from core. Time to clean up.

The change in i18n is non-conflicting. Whether the files and definitions still exist in core or not does not matter, the i18n_string implementation works regardless.

I'll update the issue title and description accordingly.

**Comment by klonos** [comment]
> ...`hook_locale`, which got removed from core

We have no change record on that 😞 ...I've created a task for us to do that at some point: #4946

**Comment by indigoxela** [comment]
@herbdool are you still interested in working on this cleanup?

Your current PR just wraps things in module_exists(), which is not what we want, because the functionality ships with i18n now.
What we need is a cleanup as described in a [previous comment](https://github.com/backdrop/backdrop-issues/issues/4838#issuecomment-778611668)

**Comment by herbdool** [comment]
@indigoxela I updated the PR based on your suggestions.

**Comment by indigoxela** [comment]
> I updated the PR based on your suggestions.

Great, many thanks for reviving this issue.

There's actually nothing to test here, as the functionality broke long time ago in early D8.
FTR this is the [related i18n issue](https://github.com/backdrop-contrib/i18n/issues/60), where all these handlers got moved to i18n - the only module using textgroups.

The code change (removal) looks good, I don't think there's anything left to do in core. RTBC :+1:

**Comment by quicksketch** [comment]
Thanks @indigoxela and @herbdool! Any chance one of you (or @klonos) would take on writing up the change log? We have a separate issue for that at https://github.com/backdrop/backdrop-issues/issues/4946.

I've merged https://github.com/backdrop/backdrop/pull/3497 into 1.x and 1.21.x. This qualifies as a "bugfix" since it's currently broken, we shouldn't have the option in core at all.
